Code/Core/Time/Timer.cpp

```diff
@@ -59,7 +64,12 @@ int64_t Timer::GetNow()
         VERIFY( QueryPerformanceCounter( &now ) );
         return now.QuadPart;
     #elif defined( __APPLE__ )
-        return (int64_t)mach_absolute_time();
+        #if defined( __ARM64__ )
+            // mach_absolute_time seems to return the wrong time on Apple Silicon
```

Comment:
From some limited testing, it looked like `mach_absolute_time` was not properly returning the elapsed time, and would cause certain tests to hang. The documentation for `mach_absolute_time` says to prefer the `clock_gettime_nsec_np` function, so that's what I'm using here for the Apple Silicon builds, which fixes the issue and makes all tests pass.

I'm not using `clock_gettime_nsec_np` on Intel Macs because that function was introduced on macOS 10.12, and I assume FASTBuild would not like to bump its macOS minspec that high. Checking only for `__ARM64__` is fine because only macOS 11.0+ can run on ARM CPUs, so `clock_gettime_nsec_np` is guaranteed to be available there.